### PR TITLE
Class and Project Views

### PR DESCRIPTION
**Pull request by alexgroomUNB**
## 🛠️ What Type of PR is This?

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Does this PR include .env changes if so message Eric or Matt (THIS IS IMPORTANT)
- [ ] Yes
- [x] No

## 📄 Description

Added a view for a class. You can and projects and they show up in list-expander view. Within each project you can expand it to view its details and teams. You can edit the description and add new teams.

## 🔗 Jira Ticket

[SCRUM-60](https://swe4103-pm-board.atlassian.net/browse/SCRUM-60)

## 🧾 Test Evidence

image
image
image
image

## 🧪 Added/Updated tests?

- [ ] Yes
- [x] No, and this is why: Once feature is fully connected with team view will test.
- [ ] I need help with writing tests

[SCRUM-60]: https://swe4103-pm-board.atlassian.net/browse/SCRUM-60?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by LucasSavoie — DISMISSED** [review]
Great job Alex! :)

**Review by aidanfoster1 — DISMISSED** [review]

File: client/src/components/AppLayout/routes.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ export const routes = {
 }
 
 export const getTitle = pathname => {
-  const key = pathname.replace('/', '')
-  return key === '' ? routes.classes.label : routes[key].label
+  const normalizedPath = pathname.replace(/^\/|\/$/g, '') // Remove leading and trailing slashes
```
Comment (matthew-collett) [inline client/src/components/AppLayout/routes.js:12]: sexy

File: client/src/views/TeamListView/TeamListView.jsx [hunk 2]
```diff
@@ -1,3 +1,183 @@
...
+import axios from 'axios'
+import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
 
-export default TeamListView
+import Button from '../../components/Button/Button'
+import Card from '../../components/Card/Card'
+import { useAuth } from '../../state/AuthProvider/AuthProvider'
+
+const ProjectView = () => {
```
Comment (matthew-collett) [inline client/src/views/TeamListView/TeamListView.jsx:9]: if this is the project view, we should rename the file to `ProjectView.jsx`. Team list view is a different screen.

Comment (alexgroomUNB) [reply client/src/views/TeamListView/TeamListView.jsx:9]: With the current logic TeamListView is not used, but may be useful later so I kept it in.

**Review by matthew-collett — CHANGES_REQUESTED** [review]
we just should be clear on what view is what and clearly define those routes to avoid confusion and ambiguity. see comments

**Pull request closed without merging**